### PR TITLE
Correctly support cluster properties

### PR DESCRIPTION
**Pull request by nishant-karajgikar**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement).

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [x] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add example if relevant.
 - [x] Document any changes to public APIs.
 - [ ] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [x] Add an entry inside this element for inclusion in the `mapbox-maps-ios` changelog: `<changelog></changelog>`.
 - [ ] Update the migration guide, API Docs, Markdown files - Readme, Developing, etc

### Summary of changes

This PR fixes the GeoJSONSource's `clusterProperties` type to [align with the style spec](https://docs.mapbox.com/mapbox-gl-js/style-spec/sources/#geojson-clusterProperties).

### CODE REVIEW

**Review by macdrevx — APPROVED** [review]

File: Sources/MapboxMaps/Style/Generated/Sources/GeojsonSource.swift [hunk 1]
```diff
@@ -67,7 +67,7 @@ Example: `{"sum": ["+", ["get", "scalerank"]]}`.
 For more advanced use cases, in place of `operator`, you can use a custom reduce expression that references a special `["accumulated"]` value, e.g.:
 `{"sum": [["+", ["accumulated"], ["get", "sum"]], ["get", "scalerank"]]}` 
     */
-    public var clusterProperties: GeoJSON?
+    public var clusterProperties: [String: Expression]?
```
Comment (mfazekas) [inline Sources/MapboxMaps/Style/Generated/Sources/GeojsonSource.swift:70]: How does this interface support `[String: [Expression]]`? See the doc comment last example above the variable declaration:
`{"sum": [["+", ["accumulated"], ["get", "sum"]], ["get", "scalerank"]]}`

Comment (nflahavan) [reply Sources/MapboxMaps/Style/Generated/Sources/GeojsonSource.swift:70]: I'm unable to create `[["+", ["accumulated"], ["get", "sum"]], ["get", "scalerank"]]` using the expression initializers. Does anyone have an example of how to do so?

Comment (nflahavan) [reply Sources/MapboxMaps/Style/Generated/Sources/GeojsonSource.swift:70]: The more I dig into this the more I feel like the correct type for `clusterProperties` should be `[String: [Expression.Element]]

Comment (macdrevx) [reply Sources/MapboxMaps/Style/Generated/Sources/GeojsonSource.swift:70]: @ZiZasaurus for visibility.

Comment (mfazekas) [reply Sources/MapboxMaps/Style/Generated/Sources/GeojsonSource.swift:70]: See #1571

**Pull request merged**